config.go

```diff
@@ -855,31 +856,20 @@ func loadConfig() (*config, []string, error) {
 		btcdLog.Warnf("%v", configFileError)
 	}
 
-	return &cfg, remainingArgs, nil
-}
-
-// btcdDial connects to the address on the named network using the appropriate
-// dial function depending on the address and configuration options.  For
-// example, .onion addresses will be dialed using the onion specific proxy if
-// one was specified, but will otherwise use the normal dial function (which
-// could itself use a proxy or not).
-func btcdDial(network, address string) (net.Conn, error) {
-	if strings.Contains(address, ".onion:") {
-		return cfg.oniondial(network, address)
+	// Provide connection manager config
+	connmgr.Dial = func(network, address string) (net.Conn, error) {
```

Comment:
Well on second look, I guess they're only be set up startup and never changed later.  Nevertheless, the package really shouldn't be using globals like this.  The code should be able to get an instance of a connection manager which houses all state.

Using globals in packages is almost always bad.